src/main/groovy/net/vivin/gradle/versioning/SemanticBuildVersioningPlugin.groovy

```diff
@@ -22,89 +22,43 @@ class SemanticBuildVersioningPlugin implements Plugin<Settings> {
-                }
-
-                if(semanticBuildVersion.promoteToRelease) {
-                    throw new BuildException('Cannot bump pre-release version when also promoting to a release version', null)
-                }
-
-                bump = VersionComponent.PRERELEASE
+                semanticBuildVersion.bump = VersionComponent.PRERELEASE
             }
 
             if(project.hasProperty('bumpPatch')) {
-                if(bump != null) {
+                if(semanticBuildVersion.bump) {
                     throw new BuildException('Cannot bump multiple version-components at the same time', null)
                 }
 
-                if(semanticBuildVersion.promoteToRelease) {
-                    throw new BuildException('Cannot bump patch-version when also promoting to a release version', null)
-                }
-
-                bump = VersionComponent.PATCH
+                semanticBuildVersion.bump = VersionComponent.PATCH
             }
 
             if(project.hasProperty('bumpMinor')) {
-                if(bump != null) {
+                if(semanticBuildVersion.bump) {
                     throw new BuildException('Cannot bump multiple version-components at the same time', null)
                 }
 
-                if(semanticBuildVersion.promoteToRelease) {
-                    throw new BuildException('Cannot bump minor-version when also promoting to a release version', null)
-                }
-
-                bump = VersionComponent.MINOR
+                semanticBuildVersion.bump = VersionComponent.MINOR
             }
 
             if(project.hasProperty('bumpMajor')) {
-                if(bump != null) {
+                if(semanticBuildVersion.bump) {
                     throw new BuildException('Cannot bump multiple version-components at the same time', null)
                 }
 
-                if(semanticBuildVersion.promoteToRelease) {
-                    throw new BuildException('Cannot bump major-version when also promoting to a release version', null)
-                }
-
-                bump = VersionComponent.MAJOR
+                semanticBuildVersion.bump = VersionComponent.MAJOR
             }
 
             if(project.hasProperty('autobump')) {
-                if(bump != null) {
-                    throw new BuildException('Cannot explicitly bump a version-component when also autobumping', null)
-                }
-
-                if(semanticBuildVersion.promoteToRelease) {
-                    throw new BuildException('Cannot explicitly promote to release when also autobumping', null)
-                }
-
-                if(semanticBuildVersion.newPreRelease) {
-                    throw new BuildException('Cannot explicitly create a new pre-release version when also autobumping', null)
-                }
-
-                semanticBuildVersion.autobump = true
+                logger.warn 'The property "autobump" is deprecated and will be ignored'
```

Comment:
Ah yes, jumping in without reading the docs (as I often do hehe). That's a good point. Let's leave it in then.